### PR TITLE
Safely borrowing SDF lexical syntax into the new parser.

### PR DESCRIPTION
**Pull request by radumereuta**
@dwightguth please review.
Should be a small fix. Some regular expressions from SDF are not supported by Java, throwing an exception even though the user doesn't care about that. In that case, just give a silent warning, and the definition would still kompile safely.

### CODE REVIEW

File: src/javasources/KTool/src/org/kframework/parser/concrete2/KSyntax2GrammarStatesFilter.java [hunk 1]
```diff
@@ -185,13 +187,24 @@ public Void visit(Production prd, Void _) {
...
+            Pattern p;
+            if (!prd.containsAttribute(Constants.REGEX)) {
+                // try to use the regular expression from SDF.
+                // If it's not compatible give a warning
+                try {
+                    p = Pattern.compile(lx.getLexicalRule());
+                } catch (PatternSyntaxException ex) {
+                    p = Pattern.compile("NoMatch");
+                    String msg = "Lexical pattern not compatible with the new parser.";
+                    GlobalSettings.kem.registerCompilerWarning(msg, lx);
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/parser/concrete2/KSyntax2GrammarStatesFilter.java]: pass the exception to this function

**Comment by dwightguth** [comment]
If you fix my comment you can merge this

File: src/javasources/KTool/src/org/kframework/parser/concrete2/KSyntax2GrammarStatesFilter.java (hunk 1 above)
Comment (radumereuta) [inline src/javasources/KTool/src/org/kframework/parser/concrete2/KSyntax2GrammarStatesFilter.java]: Didn't know about that. Nice.

**Comment by radumereuta** [comment]
This is so strange, why didn't it fail the first time?

**Comment by radumereuta** [comment]
@dwightguth please re-review. I also found an improper warning message returned by SyntaxDeclCheck.

**Pull request merged**